ts/components/helpers/withLoadingSpinner.tsx

```diff
@@ -0,0 +1,46 @@
+import hoistNonReactStatics from "hoist-non-react-statics";
+import * as React from "react";
+import Spinner, { SpinnerProps } from "react-native-loading-spinner-overlay";
+import { connect } from "react-redux";
+
+import { GlobalState } from "../../store/reducers/types";
+import { Omit } from "../../types/utils";
+
+type LoadingSpinnerInjectedProps = {
+  isLoading: boolean;
+};
+
+/**
+ * A HOC to display and overlay spinner conditionally
+ *
+ * @param WrappedComponent The react component you want to wrap
+ * @param isLoadingSelector A redux selector that returns true when the spinner need to be visible
+ * @param spinnerProps Props to pass to the spinner component
+ */
+export function withLoadingSpinner<P>(
```

Comment:
no components with `State` ?

Reply:
What you exactly mean? 

```
export function withLoadingSpinner<P, S>
```

i think we don't need it because

```
WrappedComponent: React.ComponentType<P>
```

ComponentType only needs P

Reply:
what I mean is that this HOC discards the `State` type, but perhaps that's by design since `State` is not exposed outside of the component.

Reply:
Yes exactly.